conbench/db.py

```diff
@@ -81,7 +81,7 @@ def log_after_retry_attempt(retry_state: tenacity.RetryCallState):
         "result after attempt %s for %s: %s",
         retry_state.attempt_number,
         str(retry_state.fn),
-        str(retry_state.outcome.exception()),
+        str(retry_state.outcome.exception()),  # type: ignore[union-attr]
```

Comment:
From a type inference point of view `outcome` can be `None` which is why mypy complained (no `exception()` method on `None`). Whereas as of tenacity's docstring `outcome` is always set, to the result of the future. I concluded that the current code is fine (however I am not 100 % sure because this stuff is complex, and I only thought about things; didn't try to verify with code).